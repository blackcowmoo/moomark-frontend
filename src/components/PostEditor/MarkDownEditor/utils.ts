function canManipulateViaTextNodes(targetTextArea: HTMLTextAreaElement | HTMLInputElement) {
  if (targetTextArea.nodeName !== 'TEXTAREA') {
    return false;
  }
  let browserSupportsTextareaTextNodes;
  if (typeof browserSupportsTextareaTextNodes === 'undefined') {
    const textarea = document.createElement('textarea');
    textarea.value = '1';
    browserSupportsTextareaTextNodes = !!textarea.firstChild;
  }
  return browserSupportsTextareaTextNodes;
}

export function insertText(targetTextArea: HTMLTextAreaElement | HTMLInputElement, text: string) {
  targetTextArea.focus();

  // IE 8-10
  if ((document as any).selection) {
    const ieRange = (document as any).selection.createRange();
    ieRange.text = text;

    ieRange.collapse(false /* to the end */);
    ieRange.select();

    return;
  }

  // Webkit + Edge
  const isSuccess = document.execCommand('insertText', false, text);
  if (!isSuccess) {
    const start = targetTextArea.selectionStart;
    const end = targetTextArea.selectionEnd;
    if (!(start && end)) return;
    // Firefox (non-standard method)
    if (typeof (targetTextArea as any).setRangeText === 'function') {
      (targetTextArea as any).setRangeText(text);
    } else if (canManipulateViaTextNodes(targetTextArea)) {
      const textNode = document.createTextNode(text);
      let node = targetTextArea.firstChild;

      // If textarea is empty, just insert the text
      if (!node) {
        targetTextArea.appendChild(textNode);
      } else {
        // Otherwise we need to find a nodes for start and end
        let offset = 0;
        let startNode = null;
        let endNode = null;

        // To make a change we just need a Range, not a Selection
        const range = document.createRange();

        while (node && (startNode === null || endNode === null)) {
          const nodeLength = node.nodeValue ? node.nodeValue.length
            : 0;

          // if start of the selection falls into current node
          if (start >= offset && start <= offset + nodeLength) {
            range.setStart((startNode = node), start - offset);
          }

          // if end of the selection falls into current node
          if (end >= offset && end <= offset + nodeLength) {
            range.setEnd((endNode = node), end - offset);
          }

          offset += nodeLength;
          node = node.nextSibling;
        }

        if (start !== end) {
          range.deleteContents();
        }
        range.insertNode(textNode);
      }
    } else {
      const { value } = targetTextArea;
      targetTextArea.value = value.slice(0, start) + text + value.slice(end);
    }

    targetTextArea.setSelectionRange(start + text.length, start + text.length);

    const e = document.createEvent('UIEvent');
    e.initEvent('targetTextArea', true, false);
    targetTextArea.dispatchEvent(e);
  }
}
