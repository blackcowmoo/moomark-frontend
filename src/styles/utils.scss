$global-background: #F5F5F5; 
$header-background: #ffffff;
$focus-brown: #ebc999;
// $default-green: #8ca9d3;
$default-green: #8DB48E;
$focus-red: #f26659;
$dark-green: #4D724D;
// $default-black: #000000ab;
$default-black: #373833;
$default-grey: rgb(173, 181, 189);