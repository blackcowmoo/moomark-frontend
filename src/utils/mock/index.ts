import { TilePostProps, IPostList } from 'types';
import { IUserProfile } from '@components/User/UserProfile';

const date = new Date(2021, 2, 10);
const desSample: string = '1. 주식을 사기 보다는 때를 사라.챠트는 시세의 길잡이다.2.숲을 먼저 보고 나무를 보아라.3.사는 것 보다 파는 것이 더 중요하다.4.자신에게 가장 알맞는 투자 방법을 개발하라. 5.대중이 가는 뒤안 길에 꽃길이 있다.';
export const HomePageTileMock: TilePostProps[] = [
  {
    id: 0,
    title: 'mockTitle',
    author: 'gosu',
    like: 222,
    tags: ['mockTag'],
    date,
    description: desSample,
  },
  {
    id: 0,
    title: 'ddd',
    author: 'gosu',
    like: 222,
    tags: ['mockTag'],
    date,
    description: desSample,
  },
  {
    id: 0,
    title: '도지는 신이다',
    author: '영암머스크멜론',
    like: 0,
    tags: ['mockTag'],
    date,
    description: '화성 갈꺼니까',
  },
  {
    id: 0,
    title: 'ddd',
    author: 'gosu',
    like: 222,
    tags: ['mockTag'],
    date,
    description: '123123',
  },
  {
    id: 0,
    title: 'ddd',
    author: 'gosu',
    like: 222,
    tags: ['mockTag'],
    date,
    description: '123123',
  },
  {
    id: 0,
    title: 'ddd',
    author: 'gosu',
    like: 222,
    tags: ['mockTag'],
    date,
    description: '123123',
  },
];

export const HomePageListMock: IPostList[] = [
  {
    id: 1,
    title: '투자의 책임은 너에게',
    author: 'gosu',
    like: 222,
    date,
  },
  {
    id: 2,
    title: '버진은 날아오른다',
    author: 'blackCOw',
    like: 222,
    date,
  },
  {
    id: 3,
    title: '숲을 먼저 보고 나무를 보아라.',
    author: 'gosu',
    like: 222,
    date,
  },
  {
    id: 4,
    title: '사는 것 보다 파는 것이 더 중요하다',
    author: '의문의 주식고수',
    like: 222,
    date,
  },
  {
    id: 5,
    title: 'ddd',
    author: 'gosu',
    like: 1,
    date,
  },
  {
    id: 6,
    title: 'ddd',
    author: 'gosu',
    like: 222,
    date,
  },
];

export const mockUserPageProfile: IUserProfile = {
  userName: 'test유저',
  userProfileImg: null,
  postCount: 503092,
  commentCount: 2103333,
  likeCount: 0,
};
