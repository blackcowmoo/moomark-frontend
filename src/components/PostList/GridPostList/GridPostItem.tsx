import Link from 'next/link';
import { TilePostProps } from 'types';
import { timeForToday } from 'utils/common';
import styles from './GridPostList.module.scss';

interface props {
  value: TilePostProps;
}

const GridViewItem: React.FC<props> = ({ value }) => {
  const { id, title, author, like, date, thumbnail, description } = value;
  return (
    <div className={styles.item}>
      <Link href={`/post/${id}`}>
        <a>
          {
            <div className={styles.thumbnail}>
              <img src={thumbnail || 'https://pbs.twimg.com/media/ESbTA4rUcAArJfB.jpg'} alt={thumbnail} />
            </div>
          }
        </a>
      </Link>
      <div className={styles.info}>
        <h4> {title}</h4>
        <div className={styles.description}>
          <p>{description}</p>
        </div>
        <div className={styles.subInfo}>
          <span>{timeForToday(date)}</span>
        </div>
      </div>
      <div className={styles.footer}>
        <Link href={`/user/${author}`}>
          <a className={styles.author}>
            by <b>{author}</b>
          </a>
        </Link>
        <div className={styles.like}> üçï {like}</div>
      </div>
    </div>
  );
};

export default GridViewItem;
