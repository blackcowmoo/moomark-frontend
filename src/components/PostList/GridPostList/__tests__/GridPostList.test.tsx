import { mount } from 'enzyme';
import GridPostList from '@components/PostList/GridPostList';
import { HomePageTileMock } from 'utils/mock';

describe('<GridPostList/>', () => {
  it('matches snapshot', () => {
    const wrapper = mount(<GridPostList listTitle='인기' postList={HomePageTileMock}/>);
    expect(wrapper.debug()).toMatchSnapshot();
  });
});
