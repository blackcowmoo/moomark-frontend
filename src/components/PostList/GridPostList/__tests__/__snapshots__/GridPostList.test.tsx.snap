// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GridPostList/> matches snapshot 1`] = `
"<GridPostList listTitle=\\"인기\\" postList={{...}}>
  <div className=\\"TilePostList\\">
    <h2 className=\\"listTitle\\">
      인기
    </h2>
    <div className=\\"smallWrapper\\">
      <GridViewItem value={{...}}>
        <div className=\\"item\\">
          <Link href=\\"/post/0\\">
            <a onClick={[Function: onClick]} onMouseEnter={[Function (anonymous)]} href=\\"/post/0\\">
              <div className=\\"thumbnail\\">
                <img src=\\"https://pbs.twimg.com/media/ESbTA4rUcAArJfB.jpg\\" alt={[undefined]} />
              </div>
            </a>
          </Link>
          <div className=\\"info\\">
            <h4>
               
              mockTitle
            </h4>
            <div className=\\"description\\">
              <p>
                1. 주식을 사기 보다는 때를 사라.챠트는 시세의 길잡이다.2.숲을 먼저 보고 나무를 보아라.3.사는 것 보다 파는 것이 더 중요하다.4.자신에게 가장 알맞는 투자 방법을 개발하라. 5.대중이 가는 뒤안 길에 꽃길이 있다.
              </p>
            </div>
            <div className=\\"subInfo\\">
              <span>
                2021년 3월 10일
              </span>
            </div>
          </div>
          <div className=\\"footer\\">
            <Link href=\\"/user/gosu\\">
              <a className=\\"author\\" onClick={[Function: onClick]} onMouseEnter={[Function (anonymous)]} href=\\"/user/gosu\\">
                by 
                <b>
                  gosu
                </b>
              </a>
            </Link>
            <div className=\\"like\\">
               🍕 
              222
            </div>
          </div>
        </div>
      </GridViewItem>
      <GridViewItem value={{...}}>
        <div className=\\"item\\">
          <Link href=\\"/post/0\\">
            <a onClick={[Function: onClick]} onMouseEnter={[Function (anonymous)]} href=\\"/post/0\\">
              <div className=\\"thumbnail\\">
                <img src=\\"https://pbs.twimg.com/media/ESbTA4rUcAArJfB.jpg\\" alt={[undefined]} />
              </div>
            </a>
          </Link>
          <div className=\\"info\\">
            <h4>
               
              ddd
            </h4>
            <div className=\\"description\\">
              <p>
                1. 주식을 사기 보다는 때를 사라.챠트는 시세의 길잡이다.2.숲을 먼저 보고 나무를 보아라.3.사는 것 보다 파는 것이 더 중요하다.4.자신에게 가장 알맞는 투자 방법을 개발하라. 5.대중이 가는 뒤안 길에 꽃길이 있다.
              </p>
            </div>
            <div className=\\"subInfo\\">
              <span>
                2021년 3월 10일
              </span>
            </div>
          </div>
          <div className=\\"footer\\">
            <Link href=\\"/user/gosu\\">
              <a className=\\"author\\" onClick={[Function: onClick]} onMouseEnter={[Function (anonymous)]} href=\\"/user/gosu\\">
                by 
                <b>
                  gosu
                </b>
              </a>
            </Link>
            <div className=\\"like\\">
               🍕 
              222
            </div>
          </div>
        </div>
      </GridViewItem>
      <GridViewItem value={{...}}>
        <div className=\\"item\\">
          <Link href=\\"/post/0\\">
            <a onClick={[Function: onClick]} onMouseEnter={[Function (anonymous)]} href=\\"/post/0\\">
              <div className=\\"thumbnail\\">
                <img src=\\"https://pbs.twimg.com/media/ESbTA4rUcAArJfB.jpg\\" alt={[undefined]} />
              </div>
            </a>
          </Link>
          <div className=\\"info\\">
            <h4>
               
              도지는 신이다
            </h4>
            <div className=\\"description\\">
              <p>
                화성 갈꺼니까
              </p>
            </div>
            <div className=\\"subInfo\\">
              <span>
                2021년 3월 10일
              </span>
            </div>
          </div>
          <div className=\\"footer\\">
            <Link href=\\"/user/영암머스크멜론\\">
              <a className=\\"author\\" onClick={[Function: onClick]} onMouseEnter={[Function (anonymous)]} href=\\"/user/%EC%98%81%EC%95%94%EB%A8%B8%EC%8A%A4%ED%81%AC%EB%A9%9C%EB%A1%A0\\">
                by 
                <b>
                  영암머스크멜론
                </b>
              </a>
            </Link>
            <div className=\\"like\\">
               🍕 
              0
            </div>
          </div>
        </div>
      </GridViewItem>
      <GridViewItem value={{...}}>
        <div className=\\"item\\">
          <Link href=\\"/post/0\\">
            <a onClick={[Function: onClick]} onMouseEnter={[Function (anonymous)]} href=\\"/post/0\\">
              <div className=\\"thumbnail\\">
                <img src=\\"https://pbs.twimg.com/media/ESbTA4rUcAArJfB.jpg\\" alt={[undefined]} />
              </div>
            </a>
          </Link>
          <div className=\\"info\\">
            <h4>
               
              ddd
            </h4>
            <div className=\\"description\\">
              <p>
                123123
              </p>
            </div>
            <div className=\\"subInfo\\">
              <span>
                2021년 3월 10일
              </span>
            </div>
          </div>
          <div className=\\"footer\\">
            <Link href=\\"/user/gosu\\">
              <a className=\\"author\\" onClick={[Function: onClick]} onMouseEnter={[Function (anonymous)]} href=\\"/user/gosu\\">
                by 
                <b>
                  gosu
                </b>
              </a>
            </Link>
            <div className=\\"like\\">
               🍕 
              222
            </div>
          </div>
        </div>
      </GridViewItem>
      <GridViewItem value={{...}}>
        <div className=\\"item\\">
          <Link href=\\"/post/0\\">
            <a onClick={[Function: onClick]} onMouseEnter={[Function (anonymous)]} href=\\"/post/0\\">
              <div className=\\"thumbnail\\">
                <img src=\\"https://pbs.twimg.com/media/ESbTA4rUcAArJfB.jpg\\" alt={[undefined]} />
              </div>
            </a>
          </Link>
          <div className=\\"info\\">
            <h4>
               
              ddd
            </h4>
            <div className=\\"description\\">
              <p>
                123123
              </p>
            </div>
            <div className=\\"subInfo\\">
              <span>
                2021년 3월 10일
              </span>
            </div>
          </div>
          <div className=\\"footer\\">
            <Link href=\\"/user/gosu\\">
              <a className=\\"author\\" onClick={[Function: onClick]} onMouseEnter={[Function (anonymous)]} href=\\"/user/gosu\\">
                by 
                <b>
                  gosu
                </b>
              </a>
            </Link>
            <div className=\\"like\\">
               🍕 
              222
            </div>
          </div>
        </div>
      </GridViewItem>
      <GridViewItem value={{...}}>
        <div className=\\"item\\">
          <Link href=\\"/post/0\\">
            <a onClick={[Function: onClick]} onMouseEnter={[Function (anonymous)]} href=\\"/post/0\\">
              <div className=\\"thumbnail\\">
                <img src=\\"https://pbs.twimg.com/media/ESbTA4rUcAArJfB.jpg\\" alt={[undefined]} />
              </div>
            </a>
          </Link>
          <div className=\\"info\\">
            <h4>
               
              ddd
            </h4>
            <div className=\\"description\\">
              <p>
                123123
              </p>
            </div>
            <div className=\\"subInfo\\">
              <span>
                2021년 3월 10일
              </span>
            </div>
          </div>
          <div className=\\"footer\\">
            <Link href=\\"/user/gosu\\">
              <a className=\\"author\\" onClick={[Function: onClick]} onMouseEnter={[Function (anonymous)]} href=\\"/user/gosu\\">
                by 
                <b>
                  gosu
                </b>
              </a>
            </Link>
            <div className=\\"like\\">
               🍕 
              222
            </div>
          </div>
        </div>
      </GridViewItem>
    </div>
  </div>
</GridPostList>"
`;
